File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        left = 0
        n = len(nums)
        counter = 0

        for right in range(n):
            if nums[left] == nums[right]:
                counter += 1
                if counter == 2:
                    left += 1
                    nums[left] = nums[right]
            else:
                left += 1
                nums[left] = nums[right]
                counter = 1

        return left + 1
            

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int writer = 0, count = 1;
        for(int reader = 0; reader < nums.length; reader++){
            if(nums[reader] == nums[writer]){
                count += 1;
            } else{
                count = 1;
            }

            if(count <= 2){
                writer += 1;
                nums[writer] = nums[reader];
            }
        }

        return writer + 1;
    }
}